app/email_test.go

```diff
@@ -61,3 +61,38 @@ func TestSendInviteEmailRateLimits(t *testing.T) {
 	assert.Equal(t, "app.email.rate_limit_exceeded.app_error", err.Id)
 	assert.Equal(t, http.StatusRequestEntityTooLarge, err.StatusCode)
 }
+
+func TestSendAdminUpgradeRequestEmail(t *testing.T) {
```

Comment:
Could be possible to include a test for the rate limiter error?

Reply:
Do you mean with the time included? If so, it would be difficult to test an error like so 
```
expected: "app.SendAdminUpgradeRequestEmail: Invite emails rate limit exceeded. Timer will be reset after 23h59m59.864537s seconds. Please retry after 23h59m59.864537s seconds., username=un_eanktitjd38z7ngrdk6jzkirso, retry_after_secs=86399.864537, reset_after_secs=86399.864537
```

Reply:
Yeah we should be able to test that part but taking a closer look at how it works and we need to make a bigger refactor so no need for it by now :(